instagram/src/App.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import "./App.css";
+import PostPage from "./Components/PostContainer/PostPage";
+import SearchBar from "./Components/SearchBar/SearchBar";
+import Authenticate from "./Components/Authentication/Authenticate";
+
+const PostPageAuthed = Authenticate(PostPage);
+
+class App extends Component {
+  state = {
+    username: "",
+    password: "",
+    credentials: localStorage.getItem("credentials")
+  };
+
+  handleCreds = credentials => {
+    if (!credentials) localStorage.removeItem("credentials");
+    this.setState({ credentials });
+  };
+
+  handleChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const authKey = `${this.state.username} is logged in`;
```

Comment:
it might be more useful to have `authKey = this.state.username` that way you can use it to dynamically set the username when adding comments.